Library/Homebrew/utils.rb

```diff
@@ -770,3 +770,24 @@ def number_readable(number)
   (numstr.size - 3).step(1, -3) { |i| numstr.insert(i, ",") }
   numstr
 end
+
+def truncate_text_to_approximate_size(s, max_bytes, options = {})
+  front_weight = options[:front_weight] || 0.5
+  glue = options[:glue] || "\n[...snip...]\n"
+  return s if s.bytesize <= max_bytes
+
+  if front_weight < 0 || front_weight > 1
+    raise "opts[:front_weight] must be between 0.0 and 1.0"
+  end
+  scale = 0.99 # just a tad small to allow for the glue
+  while true
```

Comment:
Looping here feels a bit weird; we should be able to calculate this so that we can calculate the maximum bytes in advance before we do any string manipulation and avoid the loop?